### ISSUE TITLE
error with `@bazel/runfiles` helper after updating to 1.6.0

### ISSUE THREAD

**Issue by supercairos** [aspect-build/rules_js#553]
After updating to 1.6.0 I have this error:

```
/private/var/tmp/_bazel_romain/aab898f10bfeb90def8d69f3c9616e24/execroot/com_shadow_cbp/bazel-out/darwin_arm64-fastbuild/bin/proxy/proxy.sh.runfiles/com_shadow_cbp/node_modules/.aspect_rules_js/@bazel+runfiles@5.7.0/node_modules/@bazel/runfiles/runfiles.js:26
            throw new Error('Every node program run under Bazel must have a $RUNFILES_DIR or $RUNFILES_MANIFEST_FILE environment variable');
            ^

Error: Every node program run under Bazel must have a $RUNFILES_DIR or $RUNFILES_MANIFEST_FILE environment variable
    at new Runfiles (/private/var/tmp/_bazel_romain/aab898f10bfeb90def8d69f3c9616e24/execroot/com_shadow_cbp/bazel-out/darwin_arm64-fastbuild/bin/proxy/proxy.sh.runfiles/com_shadow_cbp/node_modules/.aspect_rules_js/@bazel+runfiles@5.7.0/node_modules/@bazel/runfiles/runfiles.js:26:19)
    at Object.<anonymous> (/private/var/tmp/_bazel_romain/aab898f10bfeb90def8d69f3c9616e24/execroot/com_shadow_cbp/bazel-out/darwin_arm64-fastbuild/bin/proxy/proxy.sh.runfiles/com_shadow_cbp/node_modules/.aspect_rules_js/@bazel+runfiles@5.7.0/node_modules/@bazel/runfiles/index.js:9:20)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/private/var/tmp/_bazel_romain/aab898f10bfeb90def8d69f3c9616e24/execroot/com_shadow_cbp/bazel-out/darwin_arm64-fastbuild/bin/proxy/proxy.sh.runfiles/com_shadow_cbp/proxy/src/index.js:5:20)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
```

**Comment by supercairos** [comment]
Probably related to this: https://github.com/aspect-build/rules_js/pull/538

**Comment by thesayyn** [comment]
cc @gregmagolan @jbedard

**Comment by gregmagolan** [comment]
That comes from here https://github.com/bazelbuild/rules_nodejs/blob/591e76edc9ee0a71d604c5999af8bad7909ef2d4/packages/runfiles/runfiles.ts#L33

However rules_js doesn't set either the RUNFILES_DIR or the RUNFILES_MANIFEST_FILE file. Those are both set by Bazel.

Might be related to updating to Bazel 6 if you did that at the same time.

**Comment by supercairos** [comment]
I did not update to Bazel 6 :(

**Comment by gregmagolan** [comment]
Oh.. I see. It is probably because we no longer depend on the runfiles.bash initialization which got removed in #538
```
# shellcheck disable=SC1090
{

# --- begin runfiles.bash initialization v2 ---
set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
source "${RUNFILES_DIR:-/dev/null}/$f" 2>/dev/null || \
source "$(grep -sm1 "^$f " "${RUNFILES_MANIFEST_FILE:-/dev/null}" | cut -f2- -d' ')" 2>/dev/null || \
source "$0.runfiles/$f" 2>/dev/null || \
source "$(grep -sm1 "^$f " "$0.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
{ echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
# --- end runfiles.bash initialization v2 ---

}
```

**Comment by gregmagolan** [comment]
Tagging this as bug; although the fix is in rules_nodejs (https://github.com/bazelbuild/rules_nodejs/blob/stable/packages/runfiles/runfiles.ts) and not in this rule set

**Comment by supercairos** [comment]
Thanks!

**Comment by gregmagolan** [comment]
The fix for this landed in rules_nodejs: https://github.com/bazelbuild/rules_nodejs/commit/11395ea1bd86f5a3d62c1d7eaab7c1af3d0896f2

Will be available in the next release of `@bazel/runfiles`

**Comment by gregmagolan** [comment]
Keeping open until fix is verified

**Comment by jfirebaugh** [comment]
The fix in `@bazel/runfiles` 5.7.2 is working for me.

**Comment by jfirebaugh** [comment]
Sorry, I spoke too soon. Upgrading to `@bazel/runfiles` 5.7.2 and rules_js v1.8.0 had to be rolled back due to errors of the form:

```
Error: could not resolve module <runfile path>
at Runfiles.resolve (/<app>/main.sh.runfiles/<app>/node_modules/.aspect_rules_js/@bazel+runfiles@5.7.2/node_modules/@bazel/runfiles/runfiles.js:118:19)
```

**Comment by gregmagolan** [comment]
Hmm. You're getting past the `@bazel/runfiles` Runfiles constructor now with
```
        else if (!!_env['RUNFILES']) {
            this.runfilesDir = path.resolve(_env['RUNFILES']);
```
is `runfilesDir` not set to the correct value now?

**Comment by jfirebaugh** [comment]
The issue turned out to be some of our own code that set `RUNFILES_DIR`. Looks like the history was:
* We started out using rules_nodejs's `js_binary`, which generates a runfiles directories without `.sh` component, so we set `RUNFILES_DIR` accordingly for `@bazel/runfiles` to pick up.
* With rules_js@1.1.0 this value of `RUNFILES_DIR` was invalid because rules_js generates a runfiles directory with a `.sh` component, but it didn't matter because rules_js also generated a launcher script that overwrote our `RUNFILES_DIR` value with one that includes `.sh`.
* With rules_js@1.8.0 the rules_js launcher no longer overwrites `RUNFILES_DIR`, so there was a mismatch between `RUNFILES_DIR` (without `.sh`) and the actual runfiles directory (with `.sh`).
